chainer/functions/hierarchical_softmax.py

```diff
@@ -184,14 +184,16 @@ def create_huffman_tree(word_counts):
         raise ValueError('Empty vocabulary')
 
     q = six.moves.queue.PriorityQueue()
-    for w, c in six.iteritems(word_counts):
-        q.put((c, w))
+    # Use unique id when two entries have the same count.
```

Comment:
One of the suggestions is : "Add unique id to each entry so that we can compare two entries with same counts".